pkg/tfbridge/schema.go

```diff
@@ -352,28 +352,34 @@ func (ctx *conversionContext) MakeTerraformInputs(olds, news resource.PropertyMa
 
 }
 
-func (ctx *conversionContext) applyDefaults(result map[string]interface{}, olds, news resource.PropertyMap,
-	tfs shim.SchemaMap, ps map[string]*SchemaInfo, rawNames bool) error {
-
-	if !ctx.ApplyDefaults {
-		return nil
+func buildExactlyOneOfsWith(result map[string]interface{}, tfs shim.SchemaMap) map[string]struct{} {
+	exactlyOneOf := make(map[string]struct{})
+	if tfs != nil {
+		tfs.Range(func(name string, sch shim.Schema) bool {
+			if _, has := result[name]; has {
+				// `name` is present, so mark any names that are declared to
+				// exactlyOneOf with `name` for exclusion.
+				for _, exclusion := range sch.ExactlyOneOf() {
+					exactlyOneOf[exclusion] = struct{}{}
```

Comment:
You should be able to use the [StringSet](https://github.com/pulumi/pulumi/blob/v3.38.0/pkg/codegen/utilities.go#L27-L88) class for this, which would simplify the usage.

Reply:
Thanks @lblackstone - I can look at taking advantage of using StringSet for this and see what it would allow me to do